index.js

```diff
@@ -143,11 +154,14 @@ WPiPlatform.prototype.addGPIOPin = function(gpiopin) {
       .setCharacteristic(Characteristic.Model, platform.config.model ? platform.config.model : "Pi GPIO")
       .setCharacteristic(Characteristic.SerialNumber, platform.config.serial ? platform.config.serial : "Default-SerialNumber");
 
-    switch(gpiopin.mode) {
-      case "out":
+    switch(true) {
```

Comment:
For this set of case statements, I'd prefer to preserve the original code, and do something like below (which probably contains a bunch of typos / bugs!)

```
// Use "type" logic to set services
if (typeof gpiopin.type !== 'undefined') {
 switch(gpiopin.type) {
   [your new case statement]
 }
} else {
 switch (gpiopin.mode) {
 [Original Case Statement]
}
```

I'm thinking this makes it easier to check that people without updated configuration files will continue to work as anticipated - but your PR also lays good ground work I need to build into the next version of this module - on the version 2 alpha branch currently.

Hope that all makes sense!

Reply:
I leave this to you, but note I've further updated to allow additional device types. Also, though the gpio pin "mode" setting is now implicit in the device type and "mode" could be eliminated, I haven't looked into how the gpio pin settings function works;  so, though the pin "mode" should be  redundant, its left in for now for gpio pin settings purposes. "Mode" should be eliminated and the gpio pin settings determined by the type so you don't have inconsistencies between the device type and the pin settings, but I haven't figured that out yet. Perhaps you can -- or I may try some more experimenting at a later date.